### ISSUE TITLE
Download ZIP files through Data explorer

### ISSUE THREAD

**Issue by NHSBSA-Open-Data** [nhsbsa-data-analytics/Open-Data-Portal#56]
## Overview
For large file which require compression, users are unable to download files in ZIP format. The CSV_ZIP download options downloads files in .GZ format in the same way the CSV downlload option does. The expected result should match the button description. a ZIP file should be available.

Looking to create the functionality to be able to zip the multple .GX files which are created when downloading large compressed files. 

## What is affected?

![Image](https://github.com/nhsbsa-data-analytics/open-data-roadmap/assets/116061748/c363990c-272a-4872-9879-f4d6eea733cd)

## Timeframe of solution
Unkown

### What went wrong?
- 

### Why did it happen?
- 

### The solution implemented
- 

### How are we ensuring this does not happen again
-

**Comment by NHSBSA-Open-Data** [comment]
Merging two tickets: https://github.com/nhsbsa-data-analytics/Open-Data-Portal/issues/76

As discussed in a previous call, we have identified a bug with the filtered download options for compressed files.

Image

The 'Compressed CSV .ZIP option downloads a .GZ file which is not the expected output. This needs to download a compressed .ZIP file.

The option 'Compressed CSV' needs to be renamed to indicate that this will be a .GZ file. 'Compressed CSV (.GZ)'
Rename the second option 'compressed CSV .ZIP' in the similar format 'Compressed CSV (.ZIP)'

**Comment by NHSBSA-Open-Data** [comment]
@Daniellappv any update on this fix?

**Comment by sagargg** [comment]
Hi @NHSBSA-Open-Data,  This is not a bug, exporting more than 32K rows is always exported with GZIP compression. ZIP export is only available for smaller row exports. When exporting fewer than 32K rows, the data is processed on the fly in CKAN, converting it into the desired format/structure (e.g., CSV, JSON, or ZIP). However, processing larger datasets is resource intensive and can negatively impact CKAN’s performance. Therefore, for larger exports, the data is directly downloaded from BigQuery, and it does not support ZIP compression.
While it’s possible to implement ZIP downloads for large rows as well but with the asynchronous downloading, as we mentioned in the new development requirements spreadsheet.

For now, to avoid confusion, we have removed the ZIP option for large filter export and renamed the compression types to `GZIP` and `ZIP` (for smaller datasets).

<img width="245" alt="Image" src="https://github.com/user-attachments/assets/3be2f336-9d86-4175-b31d-a34fef8cb5de" />  


If you cannot see the changes, it may be due to caching. Please try opening this page in your browser and refresh the page. 
[CKAN Recline.js](https://ckan.nhs.staging.datopian.com/vendor/recline/recline.js). Please validate and let us know if we can deploy this to the production environment. 

cc  @Daniellappv @Uchechukwu-Onye-Igbo @@MuhammadIsmailShahzad @shubham-mahajan

**Comment by NHSBSA-Open-Data** [comment]
@sagargg I have tested this in staging and I'm happy with this.

However, with the new approach we are going to wait until we have a few things to test in UAT testing, complete the UAT testing and then proceed only once we know/are happy that nothing has been changed/broken unexpectedly (as things have done so in the past).

So can you add this to the next live deployment holding pen to await some deployment friends please? 

Thank you,
Kate

**Comment by sagargg** [comment]
Hi @NHSBSA-Open-Data Sure, this hasn't been deployed to production yet. We will only deploy it once you give the green light.

Can I move this to "Ready for Deployment"? If you are happy with the changes. 

cc @Daniellappv  @Uchechukwu-Onye-Igbo

**Comment by NHSBSA-Open-Data** [comment]
@sagargg Will 'Ready for Deployment' not trigger this to be deployed by anyone mistakenly, please? (just trying to cover all possibilities, apologies).

If it won't and it'll still be on hold until we've done further testing my side then yes, please, happy for you or me to change the status.

Thank you,
Kate

**Comment by sagargg** [comment]
@NHSBSA-Open-Data I'm just aksing to change the state of the issue from `testing in staging` to Ready for Deployment` so that we know that it is already tested and ready for deployment. Regarding the deployment day/time, We can agree on a day, as i mentioned in [this issue comment](https://github.com/nhsbsa-data-analytics/Open-Data-Portal/issues/52#issuecomment-2630076862).  

cc @Daniellappv @Uchechukwu-Onye-Igbo

**Comment by NHSBSA-Open-Data** [comment]
@sagargg Thank you for confirming.

Add this to the queue to await some future live launch friends please 😄

**Comment by sagargg** [comment]
Deployed on production. 
@NHSBSA-Open-Data  @Uchechukwu-Onye-Igbo

**Comment by NHSBSA-Open-Data** [comment]
Can confirm this is working as expected in live - thank you

![Image](https://github.com/user-attachments/assets/80a65f25-0802-428e-a1ba-5e6ed9338e2e)
